src/MongoDB.Driver.Core/Core/Configuration/ClusterBuilderExtensions.cs

```diff
@@ -161,16 +167,17 @@ public static ClusterBuilder ConfigureWithConnectionString(this ClusterBuilder b
             // Server
 
             // Cluster
-            builder = builder.ConfigureCluster(s => s.With(connectionMode: connectionString.Connect));
+#pragma warning disable 618
+            builder = builder.ConfigureCluster(s => s.WithConnection(connectionMode: connectionString.Connect, directConnection: connectionString.DirectConnection));
+#pragma warning restore 618
+
             if (connectionString.Hosts.Count > 0)
             {
                 builder = builder.ConfigureCluster(s => s.With(endPoints: Optional.Enumerable(connectionString.Hosts)));
             }
             if (connectionString.ReplicaSet != null)
             {
-                builder = builder.ConfigureCluster(s => s.With(
-                    connectionMode: ClusterConnectionMode.ReplicaSet,
-                    replicaSetName: connectionString.ReplicaSet));
+                builder = builder.ConfigureCluster(s => s.With(replicaSetName: connectionString.ReplicaSet));
```

Comment:
I think this assignment `connectionMode: ClusterConnectionMode.ReplicaSet,` is not required on this level. The same logic always happens in the lower level: https://github.com/DmitryLukyanov/mongo-csharp-driver/pull/135/files#diff-81e2452642fdc0ce280cdd663e9f6cb4R45-R51